lib/pharos/etcd/client.rb

```diff
@@ -0,0 +1,47 @@
+# frozen_string_literal: true
+
+require 'json'
+
+module Pharos
+  module Etcd
+    class Client
+      CURL = 'sudo curl -sSf --connect-timeout 2 --cacert /etc/pharos/pki/ca.pem --cert /etc/pharos/pki/etcd/client.pem --key /etc/pharos/pki/etcd/client-key.pem https://localhost:2379'
+
+      class Error; end
+
+      def initialize(ssh)
+        @ssh = ssh
+      end
+
+      # @return [Boolean]
+      def healthy?
+        result = @ssh.exec("#{CURL}/health")
+        return false if result.error?
+
+        data = JSON.parse(result.stdout)
+        data['health'] == 'true'
+      end
+
+      # @return [Array<Hash>]
+      def members
+        result = @ssh.exec("#{CURL}/v2/members")
+        raise Error, "Cannot fetch etcd members" if result.error?
```

Comment:
This loses the `curl` `stderr` output? Why not just use `exec!` like the other methods?

Reply:
I can change this.. reason for not using `exec!` is more clear error message.

Reply:
Yes, I imagine that having the complete `curl` command in the error message would look ugly... maybe have a `curl` helper method that raises `Error` and includes the requested URL + `result.stderr` from curl?

Reply:
Etcd client is now refactored.